### ISSUE TITLE
Map a model with deep (two levels) of has-many nested relationship

### ISSUE THREAD

**Issue by npatmaja** [scoutforpets/jsonapi-mapper#50]
Hi, I have a model that has a _has many_ relationship that also has a _has many_ relationship

```js
const A = bookshelf.Model.extend({
  b() {
    return this.hasMany(B);
  }
});

const B = bookshelf.Model.extend({
  c() {
    return this.hasMany(C);
  }
});

const C = bookshelf.Model.extend({});

// Queries
A.forge().fetch({ withRelated: ['b.c'] });
```
When I map the resulting model with jsonapi-mapper, it returns only the first level of the relationship (`A` _has many_ `B`). Is this functionality not supported yet?

_note_:
After looking at the source code, it seems that the function `mapRelations` works only for model as type `Collection` does not have property `relations` (it has property `models` tho), please do correct me if I'm wrong.

**Comment by ShadowManu** [comment]
This behavior was initially tried to be solved on #40, and it has included some tests, particularly [this one](https://github.com/scoutforpets/jsonapi-mapper/blob/master/spec/bookshelf-spec.ts#L1050) that I think touches the topic you mention. However, the solution did not completely solve the problem and its a WIP in PR #47. I'll keep working on that PR from next week when I have time available. Of course, you are more than welcome to contribute, specially on providing failing test cases ;)

**Comment by npatmaja** [comment]
Hi @ShadowManu thanks for your answer, I'll look at my case and try to create a test case (or two) out of that :)

**Comment by ShadowManu** [comment]
This has been solved on #47 and I'll go ahead of release and close this.
